### PR TITLE
[docs] [infra] add security-related headers

### PR DESCRIPTION
**Pull request by polarweasel**
Addresses SEC-37 for docs.yugabyte.com. Once we get it sorted here, I'll apply to whatever other sites I have access to.

To test, get the headers from the build preview and see if the ones this PR adds are showing up:

```sh
curl -I https://deploy-preview-10707--infallible-bardeen-164bc9.netlify.app/preview/
```

The list is:

- Strict-Transport-Security = "max-age=31536000; includeSubDomains"
- Content-Security-Policy = ""
- X-Frame-Options = "SAMEORIGIN"
- X-Content-Type-Options = "nosniff"
- Referrer-Policy = ""
- Permissions-Policy = ""
- X-XSS-Protection = "1"

### CODE REVIEW

**Comment by polarweasel** [comment]
@proff4eg Please test this out when you get a chance. And if you've got values you'd like to provide for the empty ones, please let me know what they are!

**Comment by polarweasel** [comment]
My results appear to indicate that we do need to supply values for the blank headers:

```sh
$ curl -I https://deploy-preview-10707--infallible-bardeen-164bc9.netlify.app/latest/

HTTP/2 200 
cache-control: public, max-age=0, must-revalidate
content-type: text/html; charset=UTF-8
date: Fri, 26 Nov 2021 00:37:13 GMT
etag: "09a0188124604fbd87b63d5b33ee0346-ssl"
strict-transport-security: max-age=31536000; includeSubDomains
x-xss-protection: 1
server: Netlify
x-nf-request-id: 01FNCTSGHNNM0P7D5HMRWJCMR9
x-robots-tag: noindex
age: 0
x-content-type-options: nosniff
x-frame-options: SAMEORIGIN
content-length: 0
```

File: docs/netlify.toml [hunk 1]
```diff
@@ -141,3 +141,17 @@
...
+  #
+  # https://docs.netlify.com/configure-builds/file-based-configuration/#headers
+  #
+  for = "/*"
+  [headers.values]
+    Strict-Transport-Security = "max-age=31536000; includeSubDomains"
+    Content-Security-Policy = ""
+    X-Frame-Options = "SAMEORIGIN"
+    X-Content-Type-Options = "nosniff"
+    Referrer-Policy = ""
```
Comment (proff4eg) [inline docs/netlify.toml:155]: Referrer-Policy: strict-origin

File: docs/netlify.toml [hunk 2]
```diff
@@ -141,3 +141,17 @@
...
     srcdir = "docs"
+
+[[headers]]
+  #
+  # https://docs.netlify.com/configure-builds/file-based-configuration/#headers
+  #
+  for = "/*"
+  [headers.values]
+    Strict-Transport-Security = "max-age=31536000; includeSubDomains"
+    Content-Security-Policy = ""
```
Comment (proff4eg) [inline docs/netlify.toml:152]: Content-Security-Policy: script-src 'self'

File: docs/netlify.toml [hunk 3]
```diff
@@ -141,3 +141,17 @@
...
+    X-Frame-Options = "SAMEORIGIN"
+    X-Content-Type-Options = "nosniff"
+    Referrer-Policy = ""
+    Permissions-Policy = ""
```
Comment (proff4eg) [inline docs/netlify.toml:156]: this one is a tricky one. We need to understand if the application is using any of the functionality that need to be specifically allowed by Permission Policy. Try to use this one https://www.permissionspolicy.com/  to build a sample policy.

Comment (proff4eg) [reply docs/netlify.toml:156]: use *, self or disabled

**Comment by proff4eg** [comment]
@polarweasel  I've added a few comments. But I think it's also okay to ship it in small batches. I worry that if we enable all the headers at once, the things start failing randomly and it's gonna be hard to find out which change is causing a problem.

File: docs/netlify.toml (hunk 3 above)
Comment (polarweasel) [reply docs/netlify.toml]: Using disabled for now. We can update after we've decided the other rules are working right, as you said in your comment below...

**Comment by polarweasel** [comment]
@proff4eg updated... have a look and see how it works for you now. (Also note the one still-open comment above)

**Review by haikarthikssk — CHANGES_REQUESTED** [review]

**Review by proff4eg — COMMENTED** [review]
LGTM

**Review by proff4eg — APPROVED** [review]

**Comment by polarweasel** [comment]
@haikarthikssk are we good to go on this one? I can't remember from our meeting last week

**Comment by haikarthikssk** [comment]
@polarweasel , Nope we are not. We need another call to sort this out.

**Pull request merged**